specs/kvstore/KeyValueStore.fizz

```diff
@@ -93,33 +95,30 @@ role SlateDb:
 
 
   atomic func update_memtable():
-    if self.memtable_wal_index >= self.wal_index:
-        return
-    self.memtable_wal_index = self.wal_index
     for kv in self.immutable_wal:
         self.memtable[kv[0]] = kv[1]
     
 
   atomic func freeze_memtable():
     if len(self.immutable_memtable) > 0 or len(self.memtable) == 0:
         return
-    name = "compacted/hash-" + str(self.memtable_wal_index) + ".sst"
-    if name in self.l0:
-        return
+
     self.immutable_memtable = dict(self.memtable)
     self.memtable.clear()
-    self.l0.append(name)
 
   atomic func write_l0():
     if len(self.immutable_memtable) == 0:
         return
-    name = self.l0[-1]
+    name = "compacted/ulid-" + str(next_ulid) + ".sst"
```

Comment:
Not required for this PR, but WDYT about adding a random() to the end of the string to simulate the random portion of the ULID.

Reply:
Generally we wouldn't want to add details that are not necessary for the model and adding random makes it harder as the model checker won't actually know if there is any impact off the value chosen by random number generator. 
Then, we would have to explore various possibilities of random.

Instead, here the only property we seem to depend on for ULID are,
1. Unique 
2. Lexicographical order matches chronological order

(Looking at the spec, this will not be lexicographically sorted as the next_uild goes from 9 to 10. I need to prefix with some zeros. )

Initially, I was considering adding abstract types for different [levels of measurements](https://en.wikipedia.org/wiki/Level_of_measurement). But later figured, most programmers would fallback to enums/strings for nominals, strings for ordinals and integer for interval and ratio types. So, I didn't want to complicate the specs for theoretical purity.